why-apollo-advantages-disadvantages-alternatives.md

```diff
@@ -51,93 +51,90 @@ There are plenty of popular and tech savvy companies out there who are using Apo
+Remember back when we had to trigger data fetching in one a component's lifecycle methods imperatively? Apollo Client solves this, because its data queries are declarative. React often employs a higher-order component or render prop to trigger a query automatically when a component renders. The GraphQL mutations are triggered imperatively, but that's only because a higher-order component or render prop grants access to the function which executes the mutation (e.g. on a button click). Essentially, Apollo embraces declarative programming over imperative programming.
 
 {{% sub_chapter_header "Modern State Management with GraphQL and Apollo" "apollo-state-management" %}}
 
-With the advent of GraphQL in JavaScript applications, state management entered a state (pun intended) of confusion again. Even though lots of pain points are eliminated by using a GraphQL library such as Apollo Client in your application, because it takes care of all the state management for your remote data, people are confused where to put state management libraries such as Redux or MobX now. However, it can be simple by using Redux or MobX only for local data and leave the remote data to Apollo. There is no need anymore to fetch data with asynchronous actions in Redux when Apollo Client is used for it. Thus Redux becomes the predictable state container for all your remaining application state (local data/view data/UI data, ...). Perhaps you even don't need Redux anymore, because your remaining application state hasn't enough complexity to be managed by Redux, but should be managed by React's local state instead.
+With the rise of GraphQL in JavaScript applications, state management entered another state of confusion-piun. Even though lots of pain points are eliminated using a GraphQL library like Apollo Client, since it takes care of state management for remote data, some developers are confused about where to put state management libraries like Redux or MobX now. However, it can be made simple using these libraries for local data only and leaving the remote data to Apollo. There is no longer a need to fetch data with asynchronous actions in Redux, so it becomes a predictable state container for all the remaining application state (e.g. local data/view data/UI data).  In fact, the remaining application state is simple enough to be managed by React's local state instead of Redux.
```

Comment:
confusion-piun ?

Reply:
"In fact, the remaining application state is simple enough" -> "In fact, the remaining application state may be simple enough"